pkgs/build-support/rust/build-rust-crate/test/default.nix

```diff
@@ -284,12 +293,16 @@ let
           ];
         };
         buildInputs = let
-          compile = name: text: runCommandCC name {} ''
-            mkdir -p $out/lib
-            $CC -shared -o $out/lib/${name}.so ${writeTextFile {
+          compile = name: text: let
+            src = writeTextFile {
               name = "${name}-src.c";
               inherit text;
-            }}
+            };
+          in runCommandCC name {} ''
+            mkdir -p $out/lib
+            cc -shared \
+              ${lib.optionalString stdenv.isDarwin "-undefined dynamic_lookup"} \
```

Comment:
I was part of that debugging session - surprisingly, it worked with a linux clang (which is why this probes for `stdenv.isDarwin` and not `stdenv.cc.isClang`.
Maybe we want to drop a comment about that in this file, for posterity.